src/McBopomofo.h

```diff
@@ -67,6 +67,10 @@ enum class SelectionKeys {
 FCITX_CONFIG_ENUM_NAME_WITH_I18N(SelectionKeys, N_("123456789"),
                                  N_("asdfghjkl"), N_("asdfzxcvb"));
 
+enum class CandidateLayoutHint { NotSet, Vertical, Horizontal };
```

Comment:
What about `UseSystemSetting`, i.e. uses whatever fcitx5 is using?

Reply:
Actually I found there is no system setting for candidate layout. The setting is stored in the classic UI and the Kinputpanel module and an input method module is not able to read it.

So, once a user changes the setting for the UI modules, the layout does change but the behavior of the arrow keys stays the same. It is quite weird. The only way to synchronize the key behavior and the layout is to let the input module to set the layout hint.

fcitx-chewing also has such a setting.